### PR TITLE
[WIP] Improve documentation

### PR DESCRIPTION
**Pull request by otraore**
A work in progress documentation improvement for engi

### CODE REVIEW

**Comment by EtienneBruines** [comment]
Nice to see you're working on this. 

Would you be so kind and try to use the [go guidelines](https://blog.golang.org/godoc-documenting-go-code) for this?

File: assets.go [hunk 1]
```diff
@@ -157,6 +169,8 @@ type Region struct {
 	width, height float32
 }
 
+// NewRegion Creates  new region from
```
Comment (paked) [inline assets.go]: Double space here :)

Comment (otraore) [inline assets.go]: I'll have to catch up with the new commits and then reapply some things. Thanks for spotting

**Comment by otraore** [comment]
Since this is a bit behind the master branch and I didn't completely follow the godoc conventions, let's close it for now and I'll give this a shot later on when I have time.

**Pull request closed without merging**